src/components/ListItem.js

```diff
@@ -1,17 +1,43 @@
 import React from 'react';
-import { fromMilleToHours } from '../lib/helper';
+import {
+  fromMilliSecToHours,
+  getUTCNowInMilliSec,
+  fromMilliSecToDays,
+} from '../lib/helper';
 import { shoppingLists } from '../lib/shoppingListsCollection';
+import calculateEstimate from '../lib/estimates';
 
 const ListItem = ({ listItem, listId, itemId }) => {
   const checkItem = () => {
+    const updatedItem = {
+      ...listItem,
+      recentPurchase: getUTCNowInMilliSec(),
+      previousPurchase: listItem.recentPurchase,
+      numberOfPurchases: (listItem.numberOfPurchases += 1),
+    };
+
+    if (updatedItem.previousPurchase) {
+      const latestInterval =
+        fromMilliSecToDays(updatedItem.recentPurchase) -
+        fromMilliSecToDays(updatedItem.previousPurchase);
+
+      const estimate = calculateEstimate(
```

Comment:
Hey, Sarah or Kenny! Could you be so kind and write me a little explanation on `calculateEstimate()` here? I'm not sure if I understand it correctly. 😅

Reply:
It's an algorithm🤣

Reply:
We can discuss it more on the call, but it essentially is a weighted average where the current interval between the last and more recent purchases counts a little more than the previously calculated estimated number of days until the next purchase.